contrib/aws/aws-sdk-go-v2/aws/aws.go

```diff
@@ -131,47 +135,75 @@ func (mw *traceMiddleware) startTraceMiddleware(stack *middleware.Stack) error {
 	}), middleware.After)
 }
 
-func resourceNameFromParams(requestInput middleware.InitializeInput, awsService string) (string, string, bool) {
-	var k, v string
+func resourceTagsFromParams(requestInput middleware.InitializeInput, awsService string, region string) (map[string]string, bool) {
+	tags := make(map[string]string)
 
 	switch awsService {
 	case "SQS":
-		k, v = ext.SQSQueueName, queueName(requestInput)
+		if url := queueURL(requestInput); url != "" {
+			queueName, arn := extractSQSMetadata(url, region)
+			tags[ext.SQSQueueName] = queueName
+			tags[ext.CloudResourceID] = arn
+		}
 	case "S3":
-		k, v = ext.S3BucketName, bucketName(requestInput)
+		tags[ext.S3BucketName] = bucketName(requestInput)
 	case "SNS":
-		k, v = destinationTagValue(requestInput)
+		k, v := destinationTagValue(requestInput)
+		tags[k] = v
 	case "DynamoDB":
-		k, v = ext.DynamoDBTableName, tableName(requestInput)
+		tags[ext.DynamoDBTableName] = tableName(requestInput)
 	case "Kinesis":
-		k, v = ext.KinesisStreamName, streamName(requestInput)
+		tags[ext.KinesisStreamName] = streamName(requestInput)
 	case "EventBridge":
-		k, v = ext.EventBridgeRuleName, ruleName(requestInput)
+		tags[ext.EventBridgeRuleName] = ruleName(requestInput)
 	case "SFN":
-		k, v = ext.SFNStateMachineName, stateMachineName(requestInput)
+		tags[ext.SFNStateMachineName] = stateMachineName(requestInput)
 	default:
-		return "", "", false
+		return nil, false
 	}
 
-	return k, v, true
+	return tags, true
 }
 
-func queueName(requestInput middleware.InitializeInput) string {
-	var queueURL string
+func queueURL(requestInput middleware.InitializeInput) string {
 	switch params := requestInput.Parameters.(type) {
 	case *sqs.SendMessageInput:
-		queueURL = *params.QueueUrl
+		return *params.QueueUrl
 	case *sqs.DeleteMessageInput:
-		queueURL = *params.QueueUrl
+		return *params.QueueUrl
 	case *sqs.DeleteMessageBatchInput:
-		queueURL = *params.QueueUrl
+		return *params.QueueUrl
 	case *sqs.ReceiveMessageInput:
-		queueURL = *params.QueueUrl
+		return *params.QueueUrl
 	case *sqs.SendMessageBatchInput:
-		queueURL = *params.QueueUrl
+		return *params.QueueUrl
 	}
+	return ""
+}
+
+func extractSQSMetadata(queueURL string, region string) (queueName string, arn string) {
+	// https://sqs.{region}.amazonaws.com/{account-id}/{queue-name}
 	parts := strings.Split(queueURL, "/")
-	return parts[len(parts)-1]
+	if len(parts) < 2 {
+		return "", ""
+	}
+
+	// Determine partition based on region
+	var partition string
+	switch {
+	case strings.HasPrefix(region, "cn-"):
+		partition = "aws-cn"
+	case strings.HasPrefix(region, "us-gov-"):
+		partition = "aws-us-gov"
+	default:
+		partition = "aws"
+	}
+
+	queueName = parts[len(parts)-1]
+	accountID := parts[len(parts)-2]
+
+	arn = fmt.Sprintf("arn:%s:sqs:%s:%s:%s", partition, region, accountID, queueName)
```

Comment:
Prefer `strings.Join` over `fmt.Sprintf` for maximum performance:

```suggestion
	arn = strings.Join([]string{"arn", partition, "sqs", region, accountID, queueName}, ":")
```

This might allocate but it shouldn't escape the heap.